src/what-unsafe-does.md

```diff
@@ -16,18 +16,41 @@ to your program. You definitely *should not* invoke Undefined Behavior.
 Unlike C, Undefined Behavior is pretty limited in scope in Rust. All the core
 language cares about is preventing the following things:
 
-* Dereferencing null, dangling, or unaligned pointers
-* Reading [uninitialized memory][]
+* Dereferencing (using the `*` operator on) dangling, or unaligned pointers, or
+  wide pointers with invalid metadata (see below)
 * Breaking the [pointer aliasing rules][]
-* Producing invalid primitive values:
-    * dangling/null references
-    * null `fn` pointers
+* Unwinding into another language
```

Comment:
It is currently de facto undefined to unwind out of a Rust function defined as `extern "C"` no matter the caller, because our LLVM IR generation adds `nounwind` to such functions which makes it UB to unwind out of. IIUC lang team is saying this is deliberately and explicitly UB until explicitly stated to be supported in some cases, for which there's an ongoing RFC, https://github.com/rust-lang/rfcs/pull/2699. The safeguard that aborts when you try to unwind out of an `extern "C"` Rust function did exist but was reverted twice so far because of fallout for software that relied on it despite it being UB, see https://github.com/rust-lang/rust/issues/58794 if you want the messy history and part of the discussion that led to the aforementioned RFC.